### ISSUE TITLE
Improve SDR_RA887x_Spectrum library

### ISSUE THREAD

**Issue by K7MDL2** [K7MDL2/KEITHSDR#7]
The new library form has many dependencies on externs.  

Need to make the library usage simpler and remove all externs, passing into the lib function all params (values, pointer to structures).

There are a few duplicated #defines (FFT_SIZE and RA887x display type choice).  Find a way to make these dynamic rather than #defines.

**Comment by K7MDL2** [comment]
Can now enable any combo of 4096IQm 2048IQ and 1024IQ for FFT, choose 1 of the enabled sizes for audio out and display.  Moved user data out.  Can pass preset (layout choice) and fft_size to the lib at instantiation time now.  Got rid of some pesky font lib errors, reduced duplicate font lib includes.  Documented perf results on the forum.  Still more work to do to make the API cleaner, goal is no user config editing in the lib files.

**Comment by K7MDL2** [comment]
_Because of the difficulty sharing #defines between the library and .ino,  It required editing them in 2 places each time a board revision, display type or FFT resolution was changed.  For simplicity I have merged the library back into the main program this library was originally split out form, SDR_RA887., otherwise it remains the same functionality.  The example program is useful to see how things work.
